modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,192 @@
+
+** A client for the web console called `console-test`
++
+// TODO: Should we mention anything about this being an authenticated client?
+** A client for the {oc-first} called `oc-cli-test`
+* You have access to the `kubeconfig` file generated by the installation program for the cluster.
++
+[IMPORTANT]
+====
+Before replacing the built-in OAuth server with an external provider, ensure that you have access to a valid `kubeconfig` file.
+
+If there are any issues with the external identity provider, you can use the `kubeconfig` file to gain access to the {product-title} cluster in an emergency situation.
+====
+
+.Procedure
+
+. Log in to the cluster by using your `kubeconfig` file.
++
+You must use the `kubeconfig` file to configure the direct authentication. If you use a token, the token can become invalid during the process of the procedure.
+
+. Create a secret that allows you to authenticate with the web console by running the following command:
++
+[source,terminal]
+----
+$ oc create secret generic console-secret \
+    --from-literal=clientSecret=<secret_value> \// <1>
+    -n openshift-config
+----
+<1> Replace `<secret_value>` with the value of the secret for the `console-test` client in Keycloak.
++
+////
+TODO: I skipped a step that did this:
+oc extract cm/default-ingress-cert -n openshift-config-managed --to=my-directory
+I assume we only did this because we're running keycloak from OCP, and we wouldn't do this if their provider was elsewhere, correct?
+////
+
+. Create a config map that contains the provider's certificate authority bundle by running the following command:
++
+[source,terminal]
+----
+$ oc create configmap keycloak-oidc-ca --from-file=ca-bundle.crt=my-directory/ca-bundle.crt \// <1>
+    -n openshift-config
+----
+<1> Specify the path to your provider's `ca-bundle.crt` file.
+
+. Edit the authentication configuration by running the following command:
++
+[source,terminal]
+----
+$ oc edit authentication.config/cluster
+----
+
+. Update the authentication configuration by setting the `type` field to `OIDC`, configuring the `oidcProviders` field for your Keycloak provider, and setting the `webhookTokenAuthenticator` field to `null`:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Authentication
+metadata:
+# ...
+spec:
+  oidcProviders: <1>
+  - claimMappings:
+      groups:
+        claim: groups <2>
+        prefix: 'oidc-groups-test:'
+      username:
+        claim: email <3>
+        prefixPolicy: Prefix
+        prefix:
+          prefixString: 'oidc-user-test:'
+    issuer:
+      audiences: <4>
+      - console-test
+      - oc-cli-test
+      issuerCertificateAuthority:
+        name: keycloak-oidc-ca <5>
+      issuerURL: https://keycloak-keycloak.apps.example.com/realms/master <6>
+    name: 'keycloak-oidc-server' <7>
+    oidcClients:
+    - clientID: oc-cli-test <8>
+      componentName: cli
+      componentNamespace: openshift-console
```

Comment:
@liouk @xingxingxia Just double checking that it's correct for the CLI client's namespace to also be openshift-console. If so, why is that?